templates/commands/goldentest/new_test_cli_test.go

```diff
@@ -0,0 +1,137 @@
+// limitations under the License.
+
+// Package goldentest implements golden test related subcommands.
+package goldentest
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/abcxyz/abc/templates/common"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/testutil"
+)
+
+func TestNewTestCommand(t *testing.T) {
+	t.Parallel()
+
+	specYaml := `apiVersion: 'cli.abcxyz.dev/v1beta3'
+kind: 'Template'
+
+desc: 'An example template that demonstrates the "print" action'
+
+inputs:
+  - name: 'name'
+    desc: 'the name of the person to greet'
+    default: 'Alice'
+
+steps:
+  - desc: 'Print a personalized message'
+    action: 'print'
+    params:
+      message: 'Hello, {{.name}}!'
+`
+
+	testYaml := `api_version: cli.abcxyz.dev/v1beta3
+kind: GoldenTest
+inputs:
+    - name: name
+      value: Bob
+`
+
+	cases := []struct {
+		name             string
+		newTestName      string
+		flagInputs       map[string]string
+		templateContents map[string]string
+		expectedContents map[string]string
+		wantErr          string
+	}{
+		{
+			name:        "simple_test_succeeds",
+			newTestName: "new-test",
+			flagInputs: map[string]string{
+				"name": "Bob",
+			},
+			templateContents: map[string]string{
+				"spec.yaml": specYaml,
+			},
+			expectedContents: map[string]string{
+				"test.yaml": testYaml,
+			},
+		},
+		{
+			name:        "unknown_inputs",
+			newTestName: "new-test",
+			flagInputs: map[string]string{
+				"unknown_input": "unknown",
+			},
+			templateContents: map[string]string{
+				"spec.yaml": specYaml,
+			},
+			wantErr: "unknown input(s)",
+		},
+		{
+			name:        "test_yaml_already_exists",
+			newTestName: "new-test",
+			flagInputs: map[string]string{
+				"name": "Bob",
+			},
+			templateContents: map[string]string{
+				"spec.yaml":                          specYaml,
+				"testdata/golden/new-test/test.yaml": testYaml,
+			},
+			wantErr: "can't open file",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			tempDir := t.TempDir()
+
+			common.WriteAllDefaultMode(t, tempDir, tc.templateContents)
+
+			ctx := logging.WithLogger(context.Background(), logging.TestLogger(t))
+
+			var args []string
+			args = append(args, fmt.Sprintf("--location=%s", tempDir))
+			for k, v := range tc.flagInputs {
+				args = append(args, fmt.Sprintf("--input=%s=%s", k, v))
+			}
+			args = append(args, tc.newTestName)
+
+			r := &NewTestCommand{}
+			if err := r.Run(ctx, args); err != nil {
+				if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+					t.Fatal(diff)
+				}
+				return
+			}
+
+			gotContents := common.LoadDirWithoutMode(t, filepath.Join(tempDir, "testdata/golden/", tc.newTestName))
+			if diff := cmp.Diff(gotContents, tc.expectedContents); diff != "" {
+				t.Errorf("dest directory contents were not as expected (-got,+want): %s", diff)
+			}
+		})
+	}
+}
```

Comment:
Perhaps we should also add a test for prompting. There's a test https://github.com/abcxyz/abc/blob/76f640dbba1f6519cad9e5aef6ded790851f914b/templates/common/render/render_test.go#L1116 that could be used as an example to follow.

This can be a future PR.

Reply:
Will do.

Reply:
I wrote a draft PR. But when I took a look at the original example, https://github.com/abcxyz/abc/blob/76f640dbba1f6519cad9e5aef6ded790851f914b/templates/common/render/render_test.go#L1116 is not testing against the `render` command, it is testing against `Resolve` method https://github.com/abcxyz/abc/blob/76f640dbba1f6519cad9e5aef6ded790851f914b/templates/common/render/render_test.go#L1408. Since `Resolve` has its own UT, I think we don't need to write another test the `Resolve` again in `new-test` as it has already been coverred?

Reply:
Ideally I'd like for there to be an integration test that makes sure that new-test is integrating with Resolve correctly. As long as it's not a huge effort. I may have forgotten to test this with Render() though 🫤

Reply:
I tried it but got some error of failing to read the prompt msg https://github.com/abcxyz/abc/blob/452374ecc1637863ff9ba5ef1899e503d6467cc2/templates/common/render/render_test.go#L1541 when I tested against the new-test interface. Let me try to debug what happened.